src/components/Plan.js

```diff
@@ -18,80 +18,114 @@ import React, { Component } from 'react';
-            });
+        let fields = this.props.fields;
+        let items = this.props.items;
+        let renderItem = this.props.renderItem;
 
-        let phasedIssues = {};
-        phases.forEach(phase => {
-            phasedIssues[phase] = this.props.issues.filter(issue => {
-                return issue.labels.map(label => label.name).includes(phase);
-            });
-        });
+        if (fields.length === 0) {
+            return items.sort(this.props.sortItems).map(item => renderItem(item));
+        }
+
+        let thisLevel = fields[0];
+
+        let thisField = thisLevel.field;
+        let unbucketedName = thisLevel.unbucketedName || 'noname';
+
+        let headings = [...new Set(items.filter(thisField).map(thisField))].sort(thisLevel.sort);
 
-        // Add unphased issues as an extra section
-        phasedIssues["unphased"] = this.props.issues.filter(issue => {
-            const labels = issue.labels.map(label => label.name);
-            return !labels.some(label => label.startsWith("phase:"));
+        let buckets = {};
+        headings.forEach(heading => {
+            buckets[heading] = items.filter(item => thisField(item) === heading);
         });
-        if (phasedIssues["unphased"].length > 0) {
-            phases.push("unphased");
+
+        // Put any of the items that didn't land in a headered bucket into the
+        // 'unbucketed' category.
+        let unbucketed = items.filter(item =>
+            !Object.values(buckets).reduce(Array.concat, []).includes(item));
+
+        if (unbucketed.length > 0) {
+            buckets[unbucketedName] = unbucketed;
         }
 
-        // Sort issues in all phases (including unphased) by state
-        phases.forEach(phase => {
-            phasedIssues[phase].sort((a, b) => {
-                let states = ['done', 'wip', 'todo'];
-                if (a.state === b.state) {
-                    return a.number - b.number;
-                }
-                else {
-                    return states.indexOf(a.state) - states.indexOf(b.state);
+        return (
+            Object.keys(buckets).map(bucket => {
+                if (buckets[bucket].length > 0) {
+                    return (
+                        <li className="heading" key={ bucket }>{ bucket }
+                            <ul>
+                                <IssueTree
+                                    fields={ fields.slice(1) }
+                                    items={ buckets[bucket] }
+                                    renderItem={ renderItem }
+                                    sortItems={ this.props.sortItems }
+                                />
+                            </ul>
+                        </li>
+                    )
                 }
+                else return null;
+            })
+        );
+    }
+}
+
+
+class Plan extends Component {
+
+    render() {
+        let fields = [
+            {
+                field: issue => {
```

Comment:
Maybe this should be called `label`? It seems like a `field` in one entry in the array, so it's a bit confusing for that thing to also have a key called `field` as well.

Reply:
I had another think about this and switched `fields` out for `categories` and `field` out for `label`. Does that feel any clearer?

Reply:
Yes, much clearer, thanks! 😁